### PR TITLE
[Main] Fix extension / channel not being readonly

### PR DESCRIPTION
**Pull request by MSNev**
- Bug fixes for getPlugin() not searching the TeeChannel
- Minification improcements

### CODE REVIEW

File: extensions/applicationinsights-clickanalytics-js/src/ClickAnalyticsPlugin.ts [hunk 1]
```diff
@@ -153,22 +153,15 @@ export class ClickAnalyticsPlugin extends BaseTelemetryPlugin {
...
 
-        function _initDefaults() {
-            _config = null;
-            _pageAction = null;
-            _autoCaptureHandler = null;
-            _contentHandler = null;
-            _autoCapture = false;
-
-            // Define _self.config
-            objDefineProp(self, "config", {
```
Comment (MSNev) [inline extensions/applicationinsights-clickanalytics-js/src/ClickAnalyticsPlugin.ts:166]: This was actually adding a `config` property to `window`...
Note: the `self` is missing the `_` prefix and even if it had that it's outside of the closure that defines `_self`...
So just removing as it's not used (or valid) to populate `window.config`

File: extensions/applicationinsights-properties-js/src/PropertiesPlugin.ts [hunk 2]
```diff
@@ -56,10 +56,8 @@ export default class PropertiesPlugin extends BaseTelemetryPlugin implements IPr
 
             _initDefaults();
 
-            objDefineProp(_self, "context", {
-                enumerable: true,
-                configurable: true,
-                get: function() {
+            objDefine(_self, "context", {
```
Comment (MSNev) [inline extensions/applicationinsights-properties-js/src/PropertiesPlugin.ts:59]: Using `objDefine()` as the defaults for `enumerable` and `configurable` are true and it avoids adding these uncompressible names

File: shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts [hunk 3]
```diff
@@ -65,9 +66,9 @@ const strSdkNotInitialized = "SDK is not initialized";
  */
 const defaultConfig: IConfigDefaults<IConfiguration> = objDeepFreeze({
     cookieCfg: {},
-    [STR_EXTENSIONS]: [],
-    [STR_CHANNELS]: [],
-    [STR_EXTENSION_CONFIG]: {},
+    [STR_EXTENSIONS]: { rdOnly: true, ref: true, v: [] },
```
Comment (MSNev) [inline shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts:69]: This is one of the main parts that makes the `extensions` and `channels` readonly enforced.

File: shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts [hunk 4]
```diff
@@ -791,6 +789,9 @@ export class AppInsightsCore<CfgType extends IConfiguration = IConfiguration> im
                 _configExtensions = [];
                 _channelConfig = null;
                 _channels = null;
+                let cfgChannels = _self.config.channels || [];
```
Comment (MSNev) [inline shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts:792]: This is not needed -- I tried "populating" the channels with the resolved values and it ended up duplicating them..

File: extensions/applicationinsights-analytics-js/src/JavaScriptSDK/Telemetry/PageVisitTimeManager.ts [hunk 5]
```diff
@@ -47,38 +48,23 @@ export class PageVisitTimeManager {
...
-
-                return prevPageVisitData;
-            }
-        
-            /**
-             * Starts timing visit duration of pageName
-             * @param pageName
-             * @returns {}
-             */
-            function startPageVisitTimer(pageName: string, pageUrl: string) {
```
Comment (MSNev) [inline extensions/applicationinsights-analytics-js/src/JavaScriptSDK/Telemetry/PageVisitTimeManager.ts:65]: Just removing and collapsing a function that is only used once

**Review by Karlie-777 — APPROVED** [review]

**Pull request merged**